x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/DateUtils.java

```diff
@@ -16,39 +16,88 @@
 import java.time.ZoneId;
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Locale;
 
-public class DateUtils {
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+
+public final class DateUtils {
```

Comment:
This needs to be revisited especially since the java.time support in ES has landed (see org.es.common.time package) in particular DateFormatters. Maybe there's a parser that we can reuse from there...

Reply:
Tracked here: #37703 as there are some other places we also still use joda.